src/crimson/os/alienstore/alien_store.cc

```diff
@@ -62,14 +62,44 @@ class OnCommit final: public Context
 
 namespace crimson::os {
 
-AlienStore::AlienStore(const std::string& path, const ConfigValues& values)
-  : path{path}
+static bool validate_bluestore_config(const ConfigProxy& conf)
+{
+  bool config_is_valid{true};
+  // unfortunately, the RocksDB sharding in BlueStore dramatically
+  // lifted the number of threads spawn during the RocksDB's init.
+  if (conf.get_val<bool>("bluestore_rocksdb_cf")) {
+    logger().error(
+      "AlienStore restricts bluestore_rocksdb_cf to false. "
+      "If you're using vstart.sh, pass: -o \"bluestore_rocksdb_cf=false\"");
+    config_is_valid = false;
```

Comment:
thought we could have something like
```c++
    Option("bluestore_rocksdb_cf", Option::TYPE_BOOL, Option::LEVEL_ADVANCED)
    .set_default(true)
#if defined(WITH_SEASTAR) && defined(WITH_ALIEN)
    .set_validator([](std::string *value, std::string *error_message){
      if (*value == "true" || *value != "0") {
        *error_message = "invalid BlueStore configuration. disabled";
        return -EINVAL;
      }
      return 0;
    })
#endif
    .set_description("Enable use of rocksdb column families for bluestore metadata"),
```

Reply:
It's so much better! I missed the thing and ended with reinventing a wheel. Thanks, @tchaikov!

Reply:
Dropped `defined(WITH_ALIEN)` as `options.cc` is built in single variant IIUC.